slick.grid.js

```diff
@@ -3749,12 +3782,80 @@ if (typeof Slick === "undefined") {
       selectionModel.setSelectedRanges(rowsToRanges(rows));
     }
 
-    function isGroupNode (row, cell) {
+    function isGroupNode(row, cell) {
       return $(getCellNode(row, cell))
         .parents('.slick-group')
         .length > 0;
     }
 
+    function getHiddenCssClass(index) {
+      var column = columns[index];
+      if (!column.hidden) return null;
+      if (column.showHidden) return 'show-hidden';
+      return 'hidden';
+    }
+
+    function getColumnVisibleWidth(column) {
+      return isColumnVisible(column) ? column.width : 0;
+    }
+
+    function refreshColumns() {
+      setColumns(columns);
+    }
+
+    function hideColumn(column) {
+      column.hidden = true;
+      delete(column.showHidden);
+    }
+
+    function unhideColumn(column) {
+      delete(column.hidden);
+      delete(column.showHidden);
+    }
+
+    function showHiddenColumnsByDirection(column, columnDirection) {
+      var i = getColumnIndex(column.id) + columnDirection;
+      var l = columns.length;
+      while (i >= 0 && i < l) {
+        var iteratedColumn = columns[i];
+        if (!iteratedColumn.hidden) break;
+        iteratedColumn.showHidden = true;
+        i += columnDirection;
+      }
+    }
+
+    function isColumnHidden(column) {
+      return column.hidden;
+    }
+
+    function isColumnInvisible(column) {
+      return column.hidden && !column.showHidden;
+    }
+
+    function isColumnVisible(column) {
+      return !column.hidden || column.showHidden;
+    }
+
+    function isHiddenColumnVisible(column) {
+      return column.hidden && column.showHidden;

```

Comment:
So a hidden column can be temporarily made visible by user interaction, but it's not persisted.

Hence there's this subtle difference between hidden/unhidden columns and visible/invisible columns.

Does that make sense?